lucene/core/src/java/org/apache/lucene/util/bkd/BKDReader.java

```diff
@@ -52,10 +53,17 @@
   /** Caller must pre-seek the provided {@link IndexInput} to the index location that {@link BKDWriter#finish} returned */
   public BKDReader(IndexInput in) throws IOException {
     version = CodecUtil.checkHeader(in, BKDWriter.CODEC_NAME, BKDWriter.VERSION_START, BKDWriter.VERSION_CURRENT);
-    numDims = in.readVInt();
+    int packedDims = in.readVInt();
+    if (version >= BKDWriter.VERSION_LEAF_STORES_BOUNDS) {
+      numDataDims = packedDims & 0x0000FFFF;
+      numIndexDims = packedDims >>> 16;
+    } else {
+      numDataDims = packedDims;
+      numIndexDims = numDataDims;
+    }
```

Comment:
Should we create a new version of the tree for this change and not mix it up with VERSION_LEAF_STORES_BOUNDS?